File: backend/Entities/PlaylistInvitation.cs
using System.ComponentModel.DataAnnotations;
using MassTransit;

namespace SongAppApi.Entities
{
    public class PlaylistInvitation
    {
        [Key]
        public Guid Id { get; set; } = NewId.NextSequentialGuid();

        public Guid PlaylistId { get; set; }
        public Playlist Playlist { get; set; }
        public Guid SenderId { get; set; }
        public Account Sender { get; set; }
        public Guid ReceiverId { get; set; }
        public Account Receiver { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/models/playlistinvitation.ts
import type { PlaylistVisibility } from "./playlist";

/**
 * A pending playlist-share invitation. Mirrors the backend
 * PlaylistInvitationResponse 1:1.
 *
 * Carries both sender and receiver fields so the same shape works for
 * incoming-inbox and outgoing-outbox views — the consumer picks
 * whichever side it cares about. Matches the Friendship-on-wire
 * convention used elsewhere.
 */
export interface PlaylistInvitation {
    id: string,

    playlistId: string,
    playlistName: string,
    playlistVisibility: PlaylistVisibility,

    senderId: string,
    senderUserName: string,
    senderFriendCode: string,

    receiverId: string,
    receiverUserName: string,
    receiverFriendCode: string,

    createdAt: string,
}